clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/provider/AwsInfrastructureProvider.groovy

```diff
@@ -17,39 +17,27 @@
 package com.netflix.spinnaker.clouddriver.aws.provider
 
 import com.fasterxml.jackson.core.type.TypeReference
-import com.netflix.spinnaker.cats.agent.Agent
-import com.netflix.spinnaker.cats.agent.AgentSchedulerAware
-import com.netflix.spinnaker.clouddriver.aws.AmazonCloudProvider
 import com.netflix.spinnaker.clouddriver.cache.KeyParser
 import com.netflix.spinnaker.clouddriver.cache.SearchableProvider
 import com.netflix.spinnaker.clouddriver.aws.cache.Keys
+import com.netflix.spinnaker.clouddriver.security.BaseProvider
+import org.springframework.stereotype.Component
 
 import static com.netflix.spinnaker.clouddriver.aws.cache.Keys.Namespace.SECURITY_GROUPS
-import static com.netflix.spinnaker.clouddriver.cache.SearchableProvider.SearchableResource
 
-class AwsInfrastructureProvider extends AgentSchedulerAware implements SearchableProvider {
+@Component
+class AwsInfrastructureProvider extends BaseProvider implements SearchableProvider {
```

Comment:
Same comment about not creating as `@component` but as a bean in the configuration class